Dependency injection in Oryx (#1735)
spryker/spryker-docs: _drafts/oryx/dependency-injection/oryx-service-layer.md

---
title: Oryx service layer

> description: Oryx service layer uses dependency injection to abstract system functionality

Author: ```suggestion
description: Oryx service layer leverages dependency injection
```
or maybe

```suggestion
description: Oryx service layer abstracts system functionality using DI
```
?

template: concept-topic-template
last_updated: Apr 13, 2023
---

> The service layer in Oryx serves as the foundation for the business logic, with Dependency injection(DI) playing a crucial role.

Reviewer: @dunqan

> The service layer in Oryx serves as the foundation for the business logic, with Dependency injection(DI) playing a crucial role.

Author: Isn't it described in the next two paragraphs?

Should I merge those paragraphs into one, to get rid of the `crucial` word?

> The service layer in Oryx serves as the foundation for the business logic, with Dependency injection(DI) playing a crucial role.

Reviewer: I guess, it's kind of confusing. In this sentence, we start talking about the service layer, but it's DI that plays the crucial role. And then, in next sentence, we are talking about the service layer again. 

Would it make sense to move the second part of the first sentence to the second one? For example, "The service layer in Oryx uses DI to abstract all the system functionality, including querying backend services, data caching and reloading, state management, and reactivity.

> The service layer in Oryx serves as the foundation for the business logic, with Dependency injection(DI) playing a crucial role.

Author: rephrased in one omr way, please check

The main objective of the service layer in Oryx is to abstract all the system functionality, including querying backend services, data caching and reloading, state management, and reactivity.

Furthermore, DI is utilized by many higher-level elements and concepts in Oryx, ranging from adapters and normalizers to HTTP interceptors, authentication, product loading, cart management, and checkout process.

## Dependency injection npm package

> The `@spryker-oryx/di` package provides DI not only to Oryx applications but also to other packages and can even be used outside the Oryx framework. It includes a variety of useful utilities to streamline working with DI, such as the following:

Reviewer 2: Optional suggestion:
```suggestion
The `@spryker-oryx/di` package provides DI to Oryx applications and other packages. It can even be used outside the Oryx framework. It includes a variety of useful utilities to streamline working with DI—for example:
```
